java/employee-scheduling/src/main/java/org/acme/employeescheduling/domain/Shift.java

```diff
@@ -89,6 +92,24 @@ public void setEmployee(Employee employee) {
         this.employee = employee;
     }
 
+    public boolean isOverlappingWithDate(LocalDate date) {
+        return getStart().toLocalDate().equals(date) || getEnd().toLocalDate().equals(date);
+    }
+
+    public long getOverlappingDurationInMinutes(LocalDate date) {
+        LocalDateTime startDateTime = LocalDateTime.of(date, LocalTime.MIN);
+        LocalDateTime endDateTime = LocalDateTime.of(date, LocalTime.MAX);
+        return getOverlappingDurationInMinutes(startDateTime, endDateTime, getStart(), getEnd());
+    }
+
+    private long getOverlappingDurationInMinutes(LocalDateTime firstStartDateTime, LocalDateTime firstEndDateTime,
+            LocalDateTime secondStartDateTime, LocalDateTime secondEndDateTime) {
+        LocalDateTime maxStartTime = firstStartDateTime.isAfter(secondStartDateTime) ? firstStartDateTime : secondStartDateTime;
+        LocalDateTime minEndTime = firstEndDateTime.isBefore(secondEndDateTime) ? firstEndDateTime : secondEndDateTime;
+        long minutes = maxStartTime.until(minEndTime, ChronoUnit.MINUTES);
+        return minutes > 0 ? minutes : 0;
```

Comment:
Any reason why this can't return an `int`? I doubt there will ever be 2 billion minutes of overlap. (`long` causes casting in the `penalize(...)` calls.)

Reply:
There is no reason to not return an `int`.